web-marketplace/src/components/organisms/BasicInfoForm/BasicInfoForm.tsx

```diff
@@ -144,76 +242,110 @@ const BasicInfoForm: React.FC<BasicInfoFormProps> = ({
-              value={unit}
-              emptyOptionText={_(EMPTY_OPTION_TEXT)}
-              {...form.register('regen:projectSize.qudt:unit')}
-            />
+    <>
+      <Form
+        form={form}
+        formRef={formRef}
+        isDraftRef={isDraftRef}
+        onSubmit={values => submitUntilSuccess(values, 0)}
+      >
+        <OnBoardingCard>
+          <TextField
+            label={_(BASIC_INFO_NAME_LABEL)}
+            description={_(BASIC_INFO_NAME_DESCRIPTION)}
+            placeholder={_(BASIC_INFO_NAME_PLACEHOLDER)}
+            error={!!errors['schema:name']}
+            helperText={errors['schema:name']?.message}
+            {...form.register('schema:name')}
+          />
+          <Box sx={{ display: 'flex', flexDirection: 'column', pt: [3, 12] }}>
+            <InputLabel>{_(BASIC_INFO_SIZE_LABEL)}</InputLabel>
+            <Box
+              sx={{
+                display: 'flex',
+                justifyContent: 'space-between',
+                alignItems: 'baseline',
+              }}
+            >
+              <TextField
+                className={cx(classes.parcelField, classes.parcelSize)}
+                type="number"
+                customInputProps={{ step: 'any' }}
+                sx={{ mt: { xs: 0, sm: 0 } }}
+                error={!!errors['regen:projectSize']}
+                helperText={
+                  errors['regen:projectSize']?.['qudt:numericValue']?.message
+                }
+                {...form.register('regen:projectSize.qudt:numericValue', {
+                  valueAsNumber: true,
+                })}
+              />
+              <SelectTextField
+                className={cx(classes.parcelField, classes.parcelUnit)}
+                options={[
+                  {
+                    value: 'unit:HA',
+                    label: _(msg`Hectares`),
+                  },
+                  {
+                    value: 'unit:AC',
+                    label: _(msg`Acres`),
+                  },
+                ]}
+                defaultStyle={false}
+                native={false}
+                value={unit}
+                emptyOptionText={_(EMPTY_OPTION_TEXT)}
+                {...form.register('regen:projectSize.qudt:unit')}
+              />
+            </Box>
           </Box>
-        </Box>
-      </OnBoardingCard>
-      <ProjectPageFooter
-        onPrev={onPrev}
-        isValid={isValid}
-        isSubmitting={isSubmitting}
-        dirty={isDirty}
-        saveAndExit={saveAndExit}
+        </OnBoardingCard>
+        <ProjectPageFooter
+          onPrev={onPrev}
+          isValid={isValid}
+          isSubmitting={isSubmitting}
+          dirty={isDirty}
+          saveAndExit={saveAndExit}
+        />
+      </Form>
+      <ProcessingModal
+        open={!!processingModal?.open}
+        title={_(msg`Processing transaction`)}
+        bodyText={_(msg`Please wait while the transaction is being processed.`)}
+      />
+      <TxErrorModal
```

Comment:
these modals are part of our ModalManager, which means that they are available throughout the app setting up jotai atoms: `processingModalAtom` for ProcessingModal and `errorModalAtom` and `errorCode` for TxErrorModal  so you do not need to add them here, you can just set the atoms directly